chips/cc2650/src/gpio.rs

```diff
@@ -0,0 +1,186 @@
+use core::cell::Cell;
+use core::ops::{Index, IndexMut};
+use ioc;
+use kernel::common::VolatileCell;
+use kernel::hil;
+
+const NUM_PINS: usize = 32;
+
+#[repr(C)]
+pub struct GPIO {
+    _reserved0: [u8; 0x90],
+    pub dout_set: VolatileCell<u32>,
+    _reserved1: [u8; 0xC],
+    pub dout_clr: VolatileCell<u32>,
+    _reserved2: [u8; 0xC],
+    pub dout_tgl: VolatileCell<u32>,
+    _reserved3: [u8; 0xC],
+    pub din: VolatileCell<u32>,
+    _reserved4: [u8; 0xC],
+    pub doe: VolatileCell<u32>,
+    _reserved5: [u8; 0xC],
+    pub evflags: VolatileCell<u32>,
+}
+
+pub const GPIO_BASE: usize = 0x4002_2000;
+
+#[allow(non_snake_case)]
+fn GPIO() -> &'static GPIO {
+    unsafe { &*(GPIO_BASE as *const GPIO) }
+}
+
+pub struct GPIOPin {
+    pin: usize,
+    pin_mask: u32,
+    client_data: Cell<usize>,
+    client: Cell<Option<&'static hil::gpio::Client>>,
+}
```

Comment:
I think you can restructure this to include a pointer to the registers in the ```GPIOPin struct``` instead of having a standalone function (but it's optional I have no strict opinion on it)